incubator/hnc/pkg/reconcilers/hnc_config.go

```diff
@@ -133,17 +157,25 @@ func (r *ConfigReconciler) createObjectReconcilers(inst *api.HNCConfiguration) e
 	// using the latest forest structure.
 	r.Forest.Lock()
 	defer r.Forest.Unlock()
+
+	hasError := false
 	for _, t := range inst.Spec.Types {
 		gvk := schema.FromAPIVersionAndKind(t.APIVersion, t.Kind)
 		if r.Forest.HasTypeSyncer(gvk) {
 			continue
 		}
 		if err := r.createObjectReconciler(gvk); err != nil {
-			return fmt.Errorf("Error while trying to create ObjectReconciler: %s", err)
+			msg := fmt.Sprintf("Error while trying to create ObjectReconciler for type %s: %s", gvk, err)
```

Comment:
Maybe:

```
err := r.createObjectReconciler(gvk)
if err == nil {
  continue
}
... rest of error handling code ...
```

Alternatively, pass `inst` into `createObjectReconciler` and handle the error there?

Reply:
Passed `inst` into `createObjectReconciler` to handle the error there.

Just out of curiosity... Is there any reason we prefer to handle the `nil` case? Seems to me that the opposite way (handle `err != nil` case) is quite common in our code ([example](https://github.com/kubernetes-sigs/multi-tenancy/blob/bb7eaf93524e68dc2ea42885b676b5a3d9ee728f/incubator/hnc/pkg/reconcilers/hierarchy_config.go#L242)) and in other places ([example](https://tour.golang.org/methods/19)).

Thanks so much for the suggestion!

Reply:
I prefer whichever leads to the least indented code - nothing more interesting than that. Up to you.